nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -91,38 +100,69 @@ in
 
   ###### implementation
 
-  config = mkIf (cfg.client.enable || cfg.server.enable) {
+  config =
+    let
+      wrappedSlurm = pkgs.stdenv.mkDerivation {
```

Comment:
Yes, if you look [here](https://github.com/lancelotsix/nixpkgs/commit/fe0487e540f150efd97193bf3f07664c1f43b02e#diff-cb050614c6a38ed6167757722e5ee254R133), you see that `cfg.package` is not added to `environment.systemPackages`, only `wrappedSlurm` is. So unless the administrator adds `pkgs.slurm-llnl` to the system (and then generates conflicts), `srun` will be available via the wrapper script.

The [test](https://github.com/lancelotsix/nixpkgs/commit/fe0487e540f150efd97193bf3f07664c1f43b02e#diff-a444156140a24443d8679342134312a4R77) should show that it works (in this case `root` runs `srun`, I can improve it to have a regular user run `srun` if you prefer).